### DISCUSSION TITLE
ModuleNotFoundError: No module named 'topsStack'

### DISCUSSION THREAD

**Discussion by Geo-Rser** [isce-framework/isce2#783] [General]
There was an error when i complete the installation of ISCE2.6.3 with the cuda-12.1. 
it cannot run the stackSentinel.py, the error information is:
Traceback (most recent call last):
  File "/home/ys/isce2/contrib/stack/topsStack/stackSentinel.py", line 17, in <module>
    from topsStack.Stack import config, run, sentinelSLC
ModuleNotFoundError: No module named 'topsStack'

**Comment by saoussenIris** [comment]
hi, I had the same issue and fixed it by changing the line ,in stackSentinel.py, : 
"from **topsStack.Stack** import config, run, sentinelSLC"  to
"from **Stack** import config, run, sentinelSLC"

**Comment by yunjunz** [comment]
This is a change introduced in the stack processor. You need to set up the `$PYTHONPATH` as well now, in addition to the `$PATH`, since version 2.6.2, as described here: https://github.com/isce-framework/isce2/blob/main/contrib/stack/README.md#2-setup-paths-for-stack-processors.

**Comment by Carolinah23** [comment]
Hello, I was having the same issue. 

I added the path to the `PATH` variable (`vi ~/.bashrc`, then write `export PATH="/home/...path.../contrib/stack/topsStack:${PATH}"`).
I have done this on Ubuntu 20.04 and works fine, however, in Ubuntu 22.04 it didn't. 
In Ubuntu 22.04 I applied the solution that @saoussenIris proposed and it worked (`stackSentinel.py -h` works fine). 

I rewrote "from topsStack.Stack import config, run, sentinelSLC" in `stackSentinel.py` to follow the instructions given by the ISCE team. I modified `PYTHONPATH`  (`vi ~/.bashrc`, then write `export PYTHONPATH="/home/...path.../contrib/stack/topsStack:${PYTHONPATH}"`and removed `export PATH="/home/...path.../contrib/stack/topsStack:${PATH}"`). In this case, stackSentinel.py -h gives me this error (`stackSentinel.py: command not found`).

I am wondering if this is related to the OS version or if I am modifying `PYTHONPATH` incorrectly @yunjunz ?

**Comment by yunjunz** [comment]
It should be as below. Note that there is no "topsStack" in the `PYTHONPATH` setup.

```bash
export PYTHONPATH="${PYTHONPATH}:/home/...path.../contrib/stack"
export PATH="${PATH}:/home/...path.../contrib/stack/topsStack"
```

**Reply by geofysue** [reply]
The error keeps appearing
![image](https://github.com/user-attachments/assets/06acf500-490e-4e93-97c3-62f257918651)
![image](https://github.com/user-attachments/assets/f525730a-e511-4f54-bfe9-643108dab7fe)
